### ISSUE TITLE
Run ghcide under a nix shell

### ISSUE THREAD

**Issue by moodmosaic** [moodmosaic/ide-haskell-ghcide#3]
As per @novadiscovery fork (@regnat https://github.com/novadiscovery/ide-haskell-ghcide/commit/908bfa22c5e6596da9b9207d567628365b762370)

```diff
@@ -22,7 +22,7 @@ class GhcideAutoLanguageClient
  startServerProcess(projectPath) {	  startServerProcess(projectPath) {
    let proc =	    let proc =
      (require("child_process")).spawn(	      (require("child_process")).spawn(
-        "ghcide", ["--lsp"],  { "cwd" : projectPath });
+        "nix-shell", ["--run", "ghcide --lsp"],  { "cwd" : projectPath });

    return proc;	    return proc;
  }
```

we could probably add a setting which allows to override how `ghcide` is invoked.

**Comment by thufschmitt** [comment]
Thanks for the notice :)

The fork was just a dirty experiment to try ghcide on an internal project, but the experiment seems successful, so it would be really nice to directly have this configurable rather than maintaining a fork

**Comment by moodmosaic** [comment]
It could be also as an option (checkbox) e.g. `Use nix-shell` or something(?)

**Comment by thufschmitt** [comment]
I think a text field to allow for an arbitrary command-line would be nicer as I there are other setups that might require a different command − for example the convention for running ghcide with bazel seems to be to have a `.ghcide` script in the directory (see https://github.com/tweag/rules_haskell/pull/1262).

**Comment by moodmosaic** [comment]
Perhaps two text fields, one for the actual command, and one for the comma separated args.

e.g.

```
 cmd: ghcide
args: --lsp
```

```
 cmd: nix-shell
args: --run, ghcide --lsp
```

What do you think? :smiley:

**Comment by fredericcogny** [comment]
Yes that would be lovely

**Comment by moodmosaic** [comment]
Agreed :rocket:

**Comment by moodmosaic** [comment]
@regnat, @fredericcogny, here's what we need, looks good to you? :nerd_face: 

![image](https://user-images.githubusercontent.com/287532/84036490-90a65180-a99d-11ea-8493-ba6d801df72c.png)

**Comment by thufschmitt** [comment]
Looks awesome :+1:  :)

**Comment by moodmosaic** [comment]
@regnat, if you can, it'd be nice if you can test #4 or if you can show me an easy way to test it myself. I haven't used Nix yet (perhaps now's the right time) :smiley:

**Comment by moodmosaic** [comment]
Released as ide-haskell-ghcide@v0.5.0 :taco: https://atom.io/packages/ide-haskell-ghcide

**Comment by moodmosaic** [comment]
FYI, I'm going to archive this project as per @ndmitchell's [announcement](http://neilmitchell.blogspot.com/2020/09/dont-use-ghcide-anymore-directly.html) and the [haskell-language-server](https://github.com/haskell/haskell-language-server) direction in general. I just released [`haskell` for Atom](https://atom.io/packages/haskell) with
* `haskell-language-server-wrapper` as the default
* options to run under a nix-shell, ghcide (backwards compatible with this package)

:tada:
